kr8s/_api.py

```diff
@@ -520,6 +520,11 @@ async def async_watch(
         ) as (obj_cls, response):
             async for line in response.aiter_lines():
                 event = json.loads(line)
+                if (
+                    event["object"]["kind"] == "Status"
+                    and event["object"].get("code") == 410
+                ):
+                    raise GoneError("Too old resource version")
```

Comment:
I'm not sure we want to raise a `GoneError` here. The guidance from Kubernetes is to restart the watch from the new resource version.

> clients must handle the case by recognizing the status code 410 Gone, clearing their local cache, performing a new get or list operation, and starting the watch from the resourceVersion that was returned.

Reply:
It makes sense to handle the 410 response in each watch request instead of the `wait` method, but the raise can be useful to avoid the wrong conversion (of the response in the `obj_cls`), and I don't know if there is a universal method to retrieve the new resource version ?

Reply:
We should be able to keep track of the resource version of the events. Every time we get a new event we can check and update the resource version. Every object returned from the Kubernetes API should have a `resourceVersion` so we should be able to use `obj.metadata.resourceVersion` for any object.

I expect you would need to modify this section

https://github.com/kr8s-org/kr8s/blob/ff5fad5669f6bba073148159ade600a68a34c7d6/kr8s/_api.py#L523

with something like

```python
obj = obj_cls(event["object"], api=self)
if obj.kind == kind:  # We want to ignore Status events and things
    since = obj.metadata.resourceVersion
yield event["type"], obj
```